parsl/probe/probe.py

```diff
@@ -0,0 +1,273 @@
+                if not job_st.terminal:
+                    raise Exception('The provider ({wo} launcher) did not return with terminal status'.format(wo='without' if disable_launcher else 'with'))
+
+                keyval_file = provider.channel.pull_file(os.path.expanduser('~/parsl_tmp/env.txt'), os.getcwd())
+                keyval_dict = {line.split("=")[0]: line.split("=")[1].strip() for line in open(keyval_file)}
+                process_wp = subprocess.run(['which', 'process_worker_pool.py'], stdout=subprocess.PIPE)
+
+                local_keyval = {'HOSTNAME': socket.gethostname(),
+                                'WHICHPYTHON': sys.executable,
+                                'PYTHON_VERSION': 'Python ' + platform.python_version(),
+                                'PARSL_VERSION': parsl.__version__,
+                                'PARSL_PATH': parsl.__path__[0],
+                                'ZMQ_VERSION': zmq.zmq_version(),
+                                'PYZMQ_VERSION': zmq.pyzmq_version(),
+                                'PROC_WORKER_POOL': process_wp.stdout.rstrip().decode('utf-8')}
+
+                for key, value in keyval_dict.items():
+                    if key == 'HOSTNAME':  # hostname should NOT be the same
+                        if value == local_keyval[key]:
+                            raise Exception(f'Unexpected {key}={value}')
+                    else:
+                        if not value == local_keyval[key]:
+                            raise Exception(f'Unexpected {key}={value}, expected {local_keyval[key]}')
+
+                logger.debug('\tThe provider (with launcher{wo}) returned with terminal status and consistent env variables'
+                             .format(wo=' disabled' if disable_launcher else ''))
+                shutil.rmtree(os.path.expanduser('~/parsl_tmp'))
+                os.remove('./env.txt')
+
+            except Exception:
+                raise
+
+    def check_executor(self, func, arg, expected):
+
+        # This method can also check the executors. But the uncommented one
+        # is better, bacause it does not involve dfk.
+        #
+        # dfk = parsl.load(self.config)
+        # for label, executor in dfk.executors.items():
+        #     fut = executor.submit(square, None, 2)
+        #     print(fut.result())
+
+        logger.debug('Begin checking the executor(s) in the config')
+        executors = self.config.executors
+        logger.debug(f'There are {len(executors)} executors in the config')
+
+        avail_spec = {'cores': 2, 'memory': 1000, 'disk': 1000}
+        for i, executor in enumerate(executors):
+            logger.debug(f'\tBegin checking #{i+1} executor in the config')
+            executor.provider.script_dir = '.'
+            executor.provider.channel.script_dir = '.'
+            block_id = executor.start()
+            spec = None
+            if type(executor) is WorkQueueExecutor:
+                spec = avail_spec
```

Comment:
You don't have to specify the resource_specification, just always use `None`, so you can avoid the need to import `WorkQueueExecutor`